packages/boot/src/__tests__/fixtures/stub-model-api-builder.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {bind, Component, createBindingFromClass} from '@loopback/core';
+import {
+  asModelApiBuilder,
+  ModelApiBuilder,
+  ModelApiConfig,
+} from '@loopback/model-api-builder';
+import {get, Model} from '@loopback/rest';
+import {BooterApp} from './application';
+
+@bind(asModelApiBuilder)
+class StubModelApiBuilder implements ModelApiBuilder {
+  readonly pattern: string = 'stub';
+  build(
+    application: BooterApp,
+    modelClass: typeof Model & {prototype: Model},
+    cfg: ModelApiConfig,
+  ): Promise<void> {
+    application.controller(StubController);
+    return Promise.resolve();
```

Comment:
Async functions always return a promise, there is no need to create a new empty one. Just remove this line please.

Reply:
But it's expecting `Promise<void>` to be returned and if I remove it then it'll complain that `A function whose declared type is neither 'void' nor 'any' must return a value.`

Reply:
You should define your method as `async`.

```ts
class StubModelApiBuilder implements ModelApiBuilder {
  // ...
  async build(
    application: BooterApp,
    modelClass: typeof Model & {prototype: Model},
    config: ModelApiConfig,
  ): Promise<void> {
    buildCalls.push({application, modelClass, config});
  }
}
```

Having said that, if you prefer the current style (non-async function explicitly returning a promise), then I think it's ok to use it here in tests. I would not recommend it for production code, because error handling works differently in non-async functions.

Reply:
Ah, I see. I updated it to use `async`. 😄